service/controller/aws/v14/resource/kubeconfig/current.go

```diff
@@ -0,0 +1,39 @@
+package kubeconfig
+
+import (
+	"context"
+	"fmt"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	awskey "github.com/giantswarm/cluster-operator/service/controller/aws/v14/key"
+)
+
+func (r *StateGetter) GetCurrentState(ctx context.Context, obj interface{}) ([]*corev1.Secret, error) {
+	customObject, err := awskey.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterGuestConfig := awskey.ClusterGuestConfig(customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	secretName := fmt.Sprintf("%s-kubeconfig", clusterGuestConfig.ID)
```

Comment:
Let's have a `key.KubeConfigSecretName` function for this in `pkg/v14/key` with a test.